src/main/java/com/exadel/discount/controller/VendorController.java

```diff
@@ -1,90 +1,91 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
 
 import javax.validation.Valid;
-import javax.validation.constraints.Min;
-import javax.validation.constraints.Size;
-import java.util.ArrayList;
+import javax.validation.constraints.NotNull;
 import java.util.List;
+import java.util.UUID;
 
 @Validated
 @RestController
+@RequiredArgsConstructor
 @RequestMapping("/vendors")
 public class VendorController {
-    private final List<DetailedVendorDTO> detailedVendorsServiceMock = List.of(
-            new DetailedVendorDTO(1, "Vendor1", List.of(
-                    new LocationInfoDTO(3142342, "Country", "City", "Some address", "Some contacts")
-            ), "Description")
-    );
 
-    private final List<BaseVendorDTO> baseVendorsServiceMock = List.of(
-            new BaseVendorDTO(2, "Vendor2")
-    );
+    private final VendorService vendorService;
+    private final VendorLocationService vendorLocationService;
 
     @GetMapping
     @ApiOperation("Get list of all vendors")
-    public List<BaseVendorDTO> getVendorsList() {
-        return baseVendorsServiceMock;
+    public List<VendorDTO> getVendorsList() {
+        return vendorService.getAll();
     }
 
     @GetMapping("/{vendorId}")
     @ApiOperation("Get vendor by ID")
-    public DetailedVendorDTO getVendorById(@PathVariable(name = "vendorId") @Min(value = 0) long id) {
-        return detailedVendorsServiceMock.get(0);
+    public VendorDTO getVendorById(
+            @PathVariable(name = "vendorId") @NotNull UUID id) {
+        return vendorService.getById(id);
     }
 
     @PostMapping
-    @ApiOperation("Save new vendor")
-    public DetailedVendorDTO saveNewVendor(@RequestBody @Valid CreateVendorDTO vendor) {
-        return detailedVendorsServiceMock.get(0);
+    @ApiOperation("Add new vendor")
+    public VendorDTO saveNewVendor(@RequestBody @Validated(Create.class) VendorDTO vendor) {
+        return vendorService.save(vendor);
     }
 
-    @PutMapping("/{vendorId}")
-    @ApiOperation("Update information about vendor")
-    public DetailedVendorDTO updateVendor(@PathVariable(name = "vendorId") @Min(value = 0) long vendorId,
-                                          @RequestBody @Valid DetailedVendorDTO vendor) {
-        return vendor;
+    @DeleteMapping
+    @ApiOperation("Delete vendor by ID")
+    public void deleteVendor(@RequestParam(name = "id", required = true) @NotNull UUID id) {
+        vendorService.deleteById(id);
     }
 
-    @DeleteMapping("/{vendorId}")
-    @ApiOperation("Delete vendor by ID")
-    public void deleteVendor(@PathVariable(name = "vendorId") @Min(value = 0) long vendorId) {
 
+    @GetMapping("/{vendorId}/locations")
+    @ApiOperation("Get list of all vendor locations")
+    public List<VendorLocationDTO> getVendorLocations(@PathVariable(name = "vendorId") @NotNull UUID id) {
+        return vendorService.getById(id).getVendorLocations();
```

Comment:
Why don't you get vendor locations directly from repository?

Reply:
+